sig-cli/charter.md

```diff
@@ -11,7 +11,86 @@ libraries to interface with Kubernetes API's.
 
 ### In scope
 
-SIG CLI [README]
+#### Transparently Exposing APIs through Declarative and Imperative Workflows
+
+The scope of CLI Tools focuses on enabling declarative and imperative workflows
```

Comment:
I think this could be clarified further.

kubectl provides resource-oriented commands (as opposed to some other central concept, such as package). 

kubectl is part swiss-army knife and part reference implementation for interacting with the API.

In the scope document, I mentioned that since some Kubernetes primitives are fairly low-level, in addition to general-purpose resource-oriented operations, the CLI also supports “porcelain” for common simple operational operations (both status/progress extraction and mutations) that don’t have discrete API implementations, such as run, expose, rollout, cp, top, cordon, and drain. And there should be support for non-resource-oriented APIs, such as exec, logs, attach, port-forward, and proxy.

It may be worth calling out the long-standing position that widely used functionality, such as rolling update and garbage collection, should be moved server-side, the most recent example being server-side apply. So, effectively, the CLI acts as a proving ground for new API functionality.

Reply:
But it reads more like the kubectl design principles than the SIG charter.

Maybe just say the scope is kubectl and things related to it, and then create a design principles doc separately? SIG UI calls out that its efforts center around the dashboard:
https://github.com/kubernetes/community/blob/master/sig-ui/charter.md#scope

Reply:
A subsequent but different comment: @bgrant0607 your above comments refer directly to `kubectl`.  I think this charter is saying loudly that "other cli tools" are also in scope for sig-cli.   (If we do mean "kubectl-only" then we should probably say that more explicitly)

Reply:
I am good with just kubectl.

Reply:
I would expect any new tool ideas or feature ideas need to go through a KEP. In that processes any related SIGs would need to be noted (e.g., if they did something with apps than SIG apps would need to be related). Over there we would discuss graduation criteria and situations such as has the ecosystem already handled this so something new in that space is more appropriate elsewhere (e.g., a CNCF sandbox project).

This is just me noting that some directional things are being handled by processes outside the charter.

For many of the features that move forward we should put them into the clients or move them server side so that other tools can take advantage of them. This is what happened with apply going server side and we can look to start other things there as a pattern.

Reply:
Yes.  I am a big fan of the KEP process.